src/Juvix/Compiler/Backend/Geb/Pretty/Base.hs

```diff
@@ -0,0 +1,201 @@
+doc opts x =
+  run $
+    runReader opts $
+      case atomicity x of
+        Atom -> ppCode x
+        Aggregate _ -> parens <$> ppCode x
+
+class PrettyCode c where
+  ppCode :: Member (Reader Options) r => c -> Sem r (Doc Ann)
+
+instance PrettyCode Case where
+  ppCode Case {..} = do
+    lty <- ppArg _caseLeftType
+    rty <- ppArg _caseRightType
+    cod <- ppArg _caseCodomainType
+    val <- ppArg _caseOn
+    left <- ppArg _caseLeft
+    right <- ppArg _caseRight
+    return $ kwCase <+> lty <+> rty <+> cod <+> val <+> left <+> right
+
+instance PrettyCode Pair where
+  ppCode Pair {..} = do
+    lty <- ppArg _pairLeftType
+    rty <- ppArg _pairRightType
+    left <- ppArg _pairLeft
+    right <- ppArg _pairRight
+    return $ kwPair <+> lty <+> rty <+> left <+> right
+
+instance PrettyCode Fst where
+  ppCode Fst {..} = do
+    lty <- ppArg _fstLeftType
+    rty <- ppArg _fstRightType
+    val <- ppArg _fstValue
+    return $ kwFst <+> lty <+> rty <+> val
+
+instance PrettyCode Snd where
+  ppCode Snd {..} = do
+    lty <- ppArg _sndLeftType
+    rty <- ppArg _sndRightType
+    val <- ppArg _sndValue
+    return $ kwSnd <+> lty <+> rty <+> val
+
+instance PrettyCode Lamb where
+  ppCode Lamb {..} = do
+    vty <- ppArg _lambVarType
+    bty <- ppArg _lambBodyType
+    body <- ppArg _lambBody
+    return $ kwLamb <+> vty <+> bty <+> body
+
+instance PrettyCode App where
+  ppCode App {..} = do
+    dom <- ppArg _appDomainType
+    cod <- ppArg _appCodomainType
+    left <- ppArg _appLeft
+    right <- ppArg _appRight
+    return $ kwApp <+> dom <+> cod <+> left <+> right
+
+instance PrettyCode Geb where
+  ppCode = \case
+    GebAbsurd val -> do
+      v <- ppArg val
+      return $ kwAbsurd <+> v
+    GebUnit ->
+      return kwUnit
+    GebLeft val -> do
+      v <- ppArg val
+      return $ kwLeft <+> v
+    GebRight val -> do
+      v <- ppArg val
+      return $ kwRight <+> v
+    GebCase x -> ppCode x
+    GebPair x -> ppCode x
+    GebFst x -> ppCode x
+    GebSnd x -> ppCode x
+    GebLamb x -> ppCode x
+    GebApp x -> ppCode x
+    GebVar idx -> return $ kwVar <+> annotate AnnLiteralInteger (pretty idx)
+
+instance PrettyCode Prod where
+  ppCode Prod {..} = do
+    left <- ppArg _prodLeft
+    right <- ppArg _prodRight
+    return $ kwProd <+> left <+> right
+
+instance PrettyCode Coprod where
+  ppCode Coprod {..} = do
+    left <- ppArg _coprodLeft
+    right <- ppArg _coprodRight
+    return $ kwCoprod <+> left <+> right
+
+instance PrettyCode Hom where
+  ppCode Hom {..} = do
+    dom <- ppArg _homDomain
+    cod <- ppArg _homCodomain
+    return $ kwHom <+> dom <+> cod
+
+instance PrettyCode Object where
+  ppCode = \case
+    ObjectInitial -> return kwInitial
+    ObjectTerminal -> return kwTerminal
+    ObjectProd x -> ppCode x
+    ObjectCoprod x -> ppCode x
+    ObjectHom x -> ppCode x
+
+{--------------------------------------------------------------------------------}
+{- helper functions -}
```

Comment:
Remove this

Reply:
I expect that anyone conducting a review of our codebase will not be able to distinguish the author of a specific section, reflecting our uniform and easily recognizable style that has been consistently maintained. Here I can tell is you.

Reply:
What do you want me to remove? The "{---}" or the whole comment? I can remove both, but those subdividing comments are often useful when you want to find something in a bigger file (maybe not so much in this one).

Reply:
Agree they're helpful, use the style as in other files (not with the curly braces).

Reply:
Done